File: LeetCode/Difficulty/Easy/cpp/1446.consecutive-characters.cpp
class Solution {
public:
    int maxPower(string s) {
        if (!s.length()) return 0;
        int maxP = 1, P = 1;
        for (int i = 1; i < s.length(); ++i) {
            if (s[i] == s[i-1]) {
                maxP = max(++P, maxP);
            } else {
                P = 1;
            }
        }
        return maxP;
    }
};

File: LeetCode/Difficulty/Easy/js/1446.consecutive-characters.js
/**
 * URL: https://leetcode.com/explore/featured/card/november-leetcoding-challenge/564/week-1-november-1st-november-7th/3518/
 * @param {string} s
 * @return {number}
 */
var maxPower = function(s) {
    let maxCount = 0
        , tempCount = 0
        , arr = s.split('')
        , i=0
        , len = arr.length;
    for(i=0;i<len;i++){
        let currChar = arr[i];
        let prevChar = i>0 ? arr[i-1] :false;

        if(prevChar && prevChar === currChar){
            tempCount++;
        }
        else{
            if(tempCount >= maxCount){
                maxCount = tempCount;
                tempCount = 1;
            }else{
                tempCount=1
            }
        }
    }
    return Math.max(tempCount,maxCount);

};

// To run

const output = maxPower('hooraaaaaaaaaaay');
console.log(output); // 11